x-pack/dockerlogbeat/pipelinemock/reader.go

```diff
@@ -0,0 +1,42 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package pipelinemock
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+	"io/ioutil"
+	"testing"
+
+	"github.com/docker/docker/api/types/plugins/logdriver"
+	"github.com/gogo/protobuf/proto"
+	"github.com/stretchr/testify/assert"
+)
+
+// CreateTestInput creates a mocked ReadCloser for the pipelineReader
+func CreateTestInput(t *testing.T) io.ReadCloser {
+	//setup
+	exampleStruct := &logdriver.LogEntry{
+		Source:   "Test",
+		TimeNano: 0,
+		Line:     []byte("This is a log line"),
+		Partial:  false,
+		PartialLogMetadata: &logdriver.PartialLogEntryMetadata{
+			Last:    false,
+			Id:      "",
+			Ordinal: 0,
+		},
+	}
+
+	rawBytes, err := proto.Marshal(exampleStruct)
+	assert.NoError(t, err)
+
+	sizeBytes := make([]byte, 4)
+	binary.BigEndian.PutUint32(sizeBytes, uint32(len(rawBytes)))
+	rawBytes = append(sizeBytes, rawBytes...)
+	rc := ioutil.NopCloser(bytes.NewReader(rawBytes))
+	return rc
+}
```

Comment:
The reader test checks if the line produced matches the contents hard coded here. This creates some coupling between test and this function in another package. +1 on having helpers, but the actual test should be mostly self-contained and not rely on internal knowledge of helpers. At least the `Line` should be an argument to this function. In case we want to add tests for partials or with multiple messages (e.g. for ignoring very big messages) maybe it is better to have functions like this:

```
func EncodeLog(t *testing.T, out io.Writer, entry *logdriver.Entry) {
  raw, err := proto.Marshal(entry)
  assert.NoError(t, err)
  
  sizeBytes := make([]byte, 4)
  binary.BigEndian.PutUint32(sizeBytes, uint32(len(rawBytes)))
  
  _, err := out.Write(sizeBytes)
  assert.NoError(t, err)
  _, err := out.Write(raw)
  assert.NoError(t, err)
}

func AppendTestInput(t *testing.T, out io.Writer, line string) {
  EncodeLog(t, out, &logdriver.Entry{
    ...
  })
}
```